### ISSUE TITLE
Error with generator method with a symbol name 

### ISSUE THREAD

**Issue by julien-f** [feross/standard#119]
`standard` prints `indx.js:0:0: Cannot read property 'toString' of undefined` with the following code:

```javascript
console.log(class {
  * [Symbol.iterator] () {}
})
```

There are no errors if is an object literal instead of a class:

```javascript
console.log({
  * [Symbol.iterator] () {}
})
```

And if the computed property name is not a symbol but a plain string I get the incorrect but different error message `2:12: Expected spaces around *.`:

```javascript
console.log(class {
  * ['foo'] () {}
})
```

**Comment by julien-f** [comment]
The first issue is already reported to ESpree (eslint/espree#123) but not the second one.

**Comment by feross** [comment]
Can you also report the second issue to `eslint/espree`? That's the right place for it. It's a dependency of a dependecy for `standard` ;-).

**Comment by julien-f** [comment]
After investigation it seems it simply does not work with methods in classes.

Anyways, the `generator-star` is deprecated in favor of `generator-star-spacing` (which also has an issue with generator methods but is more likely to get fixed − eslint/eslint#2351).

**Comment by feross** [comment]
I delete deprecated rules from the `.eslintrc` whenever we bump the eslint version.
